pkg/machine/e2e/config_init_test.go

```diff
@@ -71,7 +76,24 @@ func (i *initMachine) buildCmd(m *machineTestBuilder) []string {
 	if i.userModeNetworking {
 		cmd = append(cmd, "--user-mode-networking")
 	}
-	cmd = append(cmd, m.name)
+	name := m.name
+	cmd = append(cmd, name)
+
+	// when we create a new VM remove it again as cleanup
+	DeferCleanup(func() {
+		r := new(rmMachine)
+		session, err := m.setName(name).setCmd(r.withForce()).run()
+		Expect(err).ToNot(HaveOccurred(), "error occurred rm'ing machine")
+		// Some test create a invalid VM so the VM does not exists in this case we have to ignore the error.
+		// It would be much better if rm -f would behave like other commands and ignore not exists errors.
+		if session.ExitCode() == 125 {
```

Comment:
There are a lot of those tests! Would it make sense for those tests to set a `SkipMachineCleanup` or `InvalidVM` state flag?

Reply:
I am not trusting test writers to remove this flag when it is no longer needed and it doesn't help if a bug causes the machine to be created all of the sudden then we leak machines.

IMO the reasonable fix to to make rm -f not error on non existing machine like our other commands do, i.e. `podman rm -f blah`